src/main/java/core/basesyntax/lib/Injector.java

```diff
@@ -0,0 +1,38 @@
+package core.basesyntax.lib;
+
+import core.basesyntax.dao.BetDao;
+import core.basesyntax.dao.BetDaoImpl;
+import core.basesyntax.dao.UserDao;
+import core.basesyntax.dao.UserDaoImpl;
+import core.basesyntax.exception.NoSuchInjectionObjectException;
+import core.basesyntax.factory.BetDaoFactory;
+import core.basesyntax.factory.UserDaoFactory;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+
+public class Injector {
+
+    public static Object getInstance(Class clazz) throws IllegalAccessException,
+            InvocationTargetException, InstantiationException, NoSuchMethodException {
+        Constructor constructor = clazz.getDeclaredConstructor();
+        Object instance = constructor.newInstance();
+        Field[] fields = clazz.getDeclaredFields();
+        for (Field field : fields) {
+            if (field.getAnnotation(Inject.class) != null) {
+                field.setAccessible(true);
+                if (field.getType().equals(UserDao.class)
+                        && UserDaoImpl.class.isAnnotationPresent(Dao.class)) {
+                    field.set(instance, UserDaoFactory.getInstance());
+                } else if (field.getType().equals(BetDao.class)
+                        && BetDaoImpl.class.isAnnotationPresent(Dao.class)) {
+                    field.set(instance, BetDaoFactory.getInstance());
+                } else {
+                    throw new NoSuchInjectionObjectException("The required injection "
+                            + "object was not found");
+                }
```

Comment:
looks nice

I would only changed this:

```suggestion
                if (field.getType().equals(UserDao.class)
                        && UserDaoFactory.getInstance().getClass().isAnnotationPresent(Dao.class)) {
                    field.set(instance, UserDaoFactory.getInstance());
                } else if (field.getType().equals(BetDao.class)
                        && BetDaoFactory.getInstance().getClass().isAnnotationPresent(Dao.class)) {
                    field.set(instance, BetDaoFactory.getInstance());
                } else {
                    throw new NoSuchInjectionObjectException("The required injection "
                            + "object was not found");
                }
```

Reply:
done